### PR TITLE
Remove deprecated config ".global.trustDomain"

### PR DESCRIPTION
**Pull request by shankgan**
https://github.com/istio/istio/issues/27734

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/templates/configmap.yaml [hunk 1]
```diff
@@ -1,8 +1,7 @@
-
 {{- define "mesh" }}
     # The trust domain corresponds to the trust root of a system.
     # Refer to https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain
-    trustDomain: {{ .Values.global.trustDomain | default "cluster.local" | quote }}
+    trustDomain: "cluster.local"
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/templates/configmap.yaml:4]: This should be removed, we rely on the built-in defaults here

**Review by howardjohn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
When was it deprecated, i think we should keep it for at least 2 releases after that.

And also add a release note.

File: manifests/charts/istio-control/istio-discovery/templates/configmap.yaml (hunk 1 above)
Comment (shankgan) [reply manifests/charts/istio-control/istio-discovery/templates/configmap.yaml:4]: When deploying through helm, if meshConfig.trustDomain is NEVER specified OR meshConfig itself is never specified in values.yaml, will we not need this default assignment?

**Review by richardwxn — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

**Review by shamsher31 — APPROVED** [review]

**Review by ostromart — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/templates/configmap.yaml (hunk 1 above)
Comment (costinm) [reply manifests/charts/istio-control/istio-discovery/templates/configmap.yaml:4]: No - the code has it built-in.

File: manifests/charts/istio-control/istio-discovery/templates/configmap.yaml [hunk 2]
```diff
@@ -12,8 +12,6 @@
     defaultConfig:
       {{- if .Values.global.meshID }}
       meshId: {{ .Values.global.meshID }}
-      {{- else if .Values.global.trustDomain }}
```
Comment (costinm) [inline manifests/charts/istio-control/istio-discovery/templates/configmap.yaml:15]: Can you check the code (DefaultMeshConfig) and verify that if meshID is not set, we default to trustDomain ?

File: manifests/charts/istiod-remote/values.yaml [hunk 3]
```diff
@@ -173,6 +173,9 @@ meshConfig:
   # and if none are found it will search in the root namespace. Any matching declaration found in the root namespace
   # is processed as if it were declared in the leaf namespace.
   rootNamespace:
+  # The trust domain corresponds to the trust root of a system
```
Comment (costinm) [inline manifests/charts/istiod-remote/values.yaml:176]: Same as above - trustDomain default should be set in the code.

**Pull request merged**